docs: refactor pages and rewrite installation (#1014)
aneoconsulting/ArmoniK: .docs/content/1.installation/2.windows/2.wsl2-with-genie.md

# WSL2 with genie

::alert{type="warning"}
This installation method is not recommended for production environments.
::

This installation will use [`genie`](https://github.com/arkane-systems/genie). Genie is a tool which gives you a way to run systemd as pid 1. The usage of this tool will disable some Windows goodies within your Ubuntu distribution.

::alert{type="warning"}

> We does not recommend to use this installation method, even for development environment since WSL2 support natively `systemd`. We recommend to use [WSL2 with systemd](./1.wsl2-with-systemd.md).

Reviewer: ```suggestion
We do not recommend to use this installation method, even for development environment since WSL2 support natively `systemd`. We recommend to use [WSL2 with systemd](./1.wsl2-with-systemd.md).
```

::

::alert{type="info"}
You must have installed WSL2 and Ubuntu before continuing. If you haven't, please refer to the [WSL2 installation](./0.installation-using-wsl2.md) method.
::

## Install genie

To install `genie`:

```bash
cd /tmp
wget --content-disposition "https://gist.githubusercontent.com/djfdyuruiry/6720faa3f9fc59bfdf6284ee1f41f950/raw/952347f805045ba0e6ef7868b18f4a9a8dd2e47a/install-sg.sh"
chmod +x install-sg.sh
./install-sg.sh
rm install-sg.sh
```

::alert{type="info"}
Please, read the [script `install-sg.sh`](https://gist.githubusercontent.com/djfdyuruiry/6720faa3f9fc59bfdf6284ee1f41f950/raw/952347f805045ba0e6ef7868b18f4a9a8dd2e47a/install-sg.sh) before executing it.
::

Finally, in order to avoid some issues, you need to mask some services.

```bash
sudo systemctl disable getty@tty1.service multipathd.service multipathd.socket ssh.service
sudo systemctl mask systemd-remount-fs.service

> ```

Reviewer: Actually it depends on the genie version. The above script doesn't install the latest version. The latest version does not need as much masking, if any (iirc)

::alert{type="info"}
You can find more information about this issue on the [genie documentation](https://github.com/arkane-systems/genie#warning-timing-out).
::

Then, you need to exit your Ubuntu distribution and restart it.

```bash
exit
```

```powershell
wsl --shutdown
```

::alert{type="warning"}

> Please, does not start a WSL2 session with native `systemd` in parallel of `genie`. This will cause some issues with k3s when you will try to install it.

Reviewer: ```suggestion
Please, do not start a WSL2 session with native `systemd` in parallel of `genie`. This will cause some issues with k3s when you will try to install it.
```

::

## Start session

> Because `genie` must be start as PID 1, you need to start a session with `genie`.

Reviewer: ```suggestion
Because `genie` must be started as PID 1, you need to start a session with `genie`.
```

`-s` is used to initialize the bottle (if necessary), and run a shell in it. You can find more options on the [genie documentation](https://github.com/arkane-systems/genie#usage).

::alert{type="info"}

> You could also use `wsl genie -i` to start a session without initialize `genie`.

Reviewer: ```suggestion
You could also use `wsl genie -i` to initialize `genie` prior to starting a session.
```

`-s` is used to initialize the bottle (if necessary), and run a shell in it. You can find more options on the [genie documentation](https://github.com/arkane-systems/genie#usage).

::alert{type="info"}

> If you have multiple WSL2, you can specify the distribution name with `wsl -d <distro_name> genie -s`.

Reviewer: ```suggestion
If you have multiple WSL2 installations, you can specify the distribution name with `wsl -d <distro_name> genie -s`.
```

::

::alert{type="warning"}
Starting a WSL2 session with `wsl` alone will not create the session within the `genie` namespace, and thus services like Docker or Kubernetes will not behave as expected. **Be careful**.
::

> When starting you're session, you could see this :

Reviewer: ```suggestion
When starting your session, you could see this :
```

```text
Waiting for systemd....!!!!!!!!!!!!!
```

Because some services will start before `systemd`, you will see this message. You can ignore it. To start faster, you can hit `ctrl + c` to stop the waiting and re-run the command `wsl genie -s`.

::alert{type="info"}
You can find more information about this issue on the [genie documentation](https://github.com/arkane-systems/genie#warning-timing-out)
::

> 

Reviewer: Iirc, you can also use wsl genie -i first, to initialise genie. Then use wsl genie -s

## Install ArmoniK

::alert{type="warning"}

> Be carefull, you must enable `genie` support before installing ArmoniK.

Reviewer: ```suggestion
Be careful, you must enable `genie` support before installing ArmoniK.
```
